dss/api/bundles/__init__.py

```diff
@@ -31,7 +37,61 @@
 """This is the minimum amount of time remaining on the lambda for us to retry on a PUT /bundles request."""
 
 ADMIN_USER_EMAILS = set(os.environ['ADMIN_USER_EMAILS'].split(','))
+@dss_handler
+def restore(replica: str, uuid: str, version: str, confrim_code: str):
+    es_client = ElasticsearchClient.get()
+
+    # checks for valid bucket
+    if version is not None:
+        key = f"bundles/{uuid}.{version}"
+    else:
+        key = f"bundles/{uuid}"
+    bucket = Replica[replica].checkout_bucket
+    try:
+        handle = Config.get_blobstore_handle(replica)
+        handle.get(bucket=bucket, key=key)
+    except BlobNotFoundError:
+        raise DSSException(404, "not_found", "Cannot find bundle!")
+
+    if version is not None:
+        key = f"bundles/{uuid}.{version}.dead"
+    else:
+        key = f"bundles/{uuid}.dead"
 
+    try:
+        handle = Config.get_blobstore_handle(replica)
+        handle.get(bucket=bucket, key=key)
+    except BlobNotFoundError:
+        raise DSSException(404, "not_found", "Cannot find bundle!")
+
+    # Searches for any bundles with given version and uuid
+    if version is not None:
+        # matches query and deletes Specific bundle with given uuid
+        restore_key = f"{uuid}.{version}"
+        es_client.delete_by_query(
+            index="_all",
+            body={"query": {"terms": {"_id": ["{}.{}.dead".format(uuid, version)]}}},
+        )
+
+        logger.debug(f"removed dead bundle {uuid}.{version}.dead from es")
+        es_client.update_by_query(
+            index="all",
+            body={"query": {"terms": {"_id": ["{}.{}".format(uuid, version)]}}},
+        )
+        logger.debug("Restored bundle {uuid}.{version}")
+        es_back.index_bundle(restore_key)
```

Comment:
Indexing a formerly tombstoned bundle should invoke the normal DSS indexing machinery by sending an SQS message to the index queue. You can use [this snippet](https://github.com/HumanCellAtlas/data-store/blob/fbb073543290082aab4f2950c5a71858bde4ba4f/dss/operations/elasticsearch.py#L37-L39) from the manual indexing operation as an example.